cpp/examples/custom-protobuf/CMakeLists.txt

```diff
@@ -0,0 +1,88 @@
+cmake_minimum_required(VERSION 3.14)
+project(quickstart LANGUAGES CXX)
+
+# Set C++ standard
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Find Protocol Buffers
+set(Protobuf_DEBUG ON)  # Enable debug output
+set(Protobuf_USE_STATIC_LIBS ON)  # Force static libraries
+set(Protobuf_VERSION 3.21.12)  # Explicitly set version to match system library
```

Comment:
Maybe we should containerize this example in particular? I think we could probably install protobuf as part of the build, but keeping the configuration smaller also seems nice.